static/src/javascripts/projects/common/modules/article/space-filler.js

```diff
@@ -2,47 +2,57 @@ define([
     'common/utils/fastdom-promise',
     'Promise',
     'raven',
-    'common/modules/article/spacefinder'
+    'common/modules/article/spacefinder',
+    'common/modules/commercial/dfp/QueueAsync'
```

Comment:
I thought about this. Have you checked if memory is released in chained promises, once the promise is resolved? Might be good to have a look at the polyfill, as chaining many times can have bad results. Alternatively a simple FIFO queue where you override the `push` method might be a better alternative, especially since we're not really interested in the resolved value of the queueing process itself.